app/Http/Controllers/PhotoController.php

```diff
@@ -0,0 +1,96 @@
+{
+    private $_PhotoAlbumRepository;
+    private $_PhotoRepository;
+
+    public function __construct(RepositorieFactory $repositorieFactory){
+        $this->_PhotoAlbumRepository = $repositorieFactory->getRepositorie(RepositorieFactory::$PHOTOALBUMREPOKEY);
+        $this->_PhotoRepository = $repositorieFactory->getRepositorie(RepositorieFactory::$PHOTOREPOKEY);
+        $this->middleware('auth');
+    }
+
+    public function index($id)
+    {
+        PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->find($id);
+        $photoAlbum->description = str_replace("\r\n","<br>", $photoAlbum->description);
+        $photos = $this->getPhotos($id); 
+        $curPageName = $photoAlbum->title;
+        return view('front-end.photo_album', compact('curPageName', 'photoAlbum', 'photos'));
+    }
+
+    //List of photos.
+    public function getPhotos($id){
+        PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->find($id);
+        if($photoAlbum != null){
+            $photosObjects = $this->_PhotoAlbumRepository->getThumbnails($photoAlbum->id);
+            return $photosObjects;
+        } else{
+            abort(404);
+        }
+    }
+
+    public function addAlbum(Request $request){
+        $title = $request->title;
+        $description = $request->description;
+        $date = $request->captureDate;
+        if($title!= null && $description !=null && $date != null){
+            if(strlen($title) < 256 && strlen($description) < 256 ){
+                PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->create(["title" => $title, "description"=> $description, "date" => $date]);
+                $this->addPhotoToAlbum($request, $photoAlbum->id);
+                return $photoAlbum->id;
+            } else{
+                redirect()->back()->withErrors(["error" => trans('front-end/photo.inputToLong')]);
+            }
+        } 
+    }  
+
+    public function addPhotoToAlbum(Request $request, $ablumId){
+        //get title from given album
+        PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->find($ablumId);
+        $photos = $request->photos;
+        $thumbnails = $request->thumbnails;
+        if($photos !=null && $thumbnails  !=null){
+            for ($i = 0; $i < count($photos); $i++) {
+                $photo = $photos[$i];
+                $thumbnail = $thumbnails[$i];
+                $fileExtension = $photos[$i]->clientExtension();
+                if($fileExtension == "png"||"jpeg"||"jpg"||"gif"||"svg"){  
+                    $this->savePhoto($photo, $photoAlbum, $thumbnail, $fileExtension);
+                }
+            }                        
+        }
+        return redirect()->route('PhotoAlbum', ['albumId' => $ablumId]);
+    }
+
+    public function savePhoto($image, $photoAlbum, $thumbnail, $fileExtension){
+        Photo:$photo = $this->_PhotoRepository->create(["album" => $photoAlbum]);
+        $imageFileName = $photo->id . '.' . $fileExtension;
```

Comment:
Het lijkt me beter om de foto's een naam te geven die bestaat uit >=20 random characters. De aws link naar de foto's is namelijk publiek, en op deze manier is het relatief makkelijk all foto's op te halen. Hetzelfde geldt voor de thumbnail naam.

Reply:
Willen we niet gewoon dat de foto's openbaar zijn? Is wel mooi om aan potentiële leden te kunnen laten zien wat de esac doet.

Reply:
Momenteel kunnen via de website alleen ingelogde gebruikers de foto's zien. Ik weet niet hoe het verder zit met foto's en GDPR.

Reply:
Volgens de privacy policy van de esac is het geen probleem.

Reply:
Dan lijkt het me prima zo.